Woche4/filter-more/helpers.c

```diff
@@ -0,0 +1,182 @@
+            image[i][j].rgbtRed = image[i][width - j - 1].rgbtRed;
+
+            image[i][width - j - 1].rgbtBlue = temp[0];
+            image[i][width - j - 1].rgbtGreen = temp[1];
+            image[i][width - j - 1].rgbtRed = temp[2];
+
+        }
+    }
+
+    return;
+}
+
+int getBlur(int i, int j, int height, int width, RGBTRIPLE image[height][width] , int color_position) {
+    float counter = 0;
+    int sum = 0;
+
+    for (int k = i - 1; k <  (i + 2); k++) {
+        for (int l = j - 1; l < (j + 2); l ++) {
+            if(k < 0 || l < 0 || k >= height || l >= width) {
+                continue;
+            }
+            if (color_position == 0) {
+                sum += image[k][l].rgbtRed;
+            }
+            else if (color_position == 1) {
+                sum += image[k][l].rgbtGreen;
+            }
+            else {
+                sum += image[k][l].rgbtBlue;
+            }
+            counter++;
+
+        }
+    }
+    return round(sum /counter);
+}
+
+void blur(int height, int width, RGBTRIPLE image[height][width]) {
+    RGBTRIPLE copy[height][width];
+
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            copy[i][j] = image[i][j];
+        }
+    }
+
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            image[i][j].rgbtRed = getBlur(i, j, height, width, copy, 0);
+            image[i][j].rgbtGreen = getBlur(i, j, height, width, copy, 1);
+            image[i][j].rgbtBlue = getBlur(i, j, height, width, copy, 2);
+        }
+    }
+}
+
+void edges(int height, int width, RGBTRIPLE image[height][width]) {
+    RGBTRIPLE temp[height][width];
+         
+    int GxR;
+    int GyR;
+    int GxG;
+    int GyG;
+    int GxB;
+    int GyB;
+        
+    int Gx[3][3] = {{-1, 0, 1}, {-2, 0, 2}, {-1, 0, 1}};
+    int Gy[3][3] = {{-1, -2, -1}, {0, 0, 0}, {1, 2, 1}};
+        
+    for (int i = 0; i < height; i++) {
+        for (int j = 0; j < width; j++) {
+            GxR = 0;
+            GyR = 0;
+            GxG = 0;
+            GyG = 0;
+            GxB = 0;
+            GyB = 0;
+    
+            for (int h = -1; h < 2; h++) {
+                for (int w = -1; w < 2; w++) {
+                    if (i + h < 0 || i + h > height - 1) {
+                        continue;
+                    }
+                        
+                    if (j + w < 0 || j + w > width - 1) {
+                        continue;
+                    }
+
+                    GxR += image[i + h][j + w].rgbtRed * Gx[h + 1][w + 1];
+                    GxG += image[i + h][j + w].rgbtGreen * Gx[h + 1][w + 1];
+                    GxB += image[i + h][j + w].rgbtBlue * Gx[h + 1][w + 1];
+
+                    GyR += image[i + h][j + w].rgbtRed * Gy[h + 1][w + 1];
+                    GyG += image[i + h][j + w].rgbtGreen * Gy[h + 1][w + 1];
+                    GyB += image[i + h][j + w].rgbtBlue * Gy[h + 1][w + 1];
+                }
```

Comment:
```suggestion
                   RGBTRIPLE imageTripple = image[i + h][j + w];
                    GxR += imageTriple.rgbtRed * Gx[h + 1][w + 1];
                    GxG += imageTriple.rgbtGreen * Gx[h + 1][w + 1];
                    GxB += imageTriple.rgbtBlue * Gx[h + 1][w + 1];

                    GyR += imageTriple.rgbtRed * Gy[h + 1][w + 1];
                    GyG += imageTriple.rgbtGreen * Gy[h + 1][w + 1];
                    GyB += imageTriple.rgbtBlue * Gy[h + 1][w + 1];
                }
```

by using a variable in this case `imageTriple` you 'iterate' through the array once and not many times.